client/components/data/query-purchase-cancellation-offers/index.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchCancellationOffers } from 'calypso/state/cancellation-offers/actions';
+import isFetchingCancellationOffers from 'calypso/state/cancellation-offers/selectors/is-fetching-cancellation-offers';
+
+interface OwnProps {
+	siteId?: number;
+	purchaseId?: number;
+}
+
+const QueryPurchaseCancellationOffers: React.FC< OwnProps > = ( { siteId, purchaseId } ) => {
+	const dispatch = useDispatch();
+
+	const fetchingCancellationOffers = useSelector( ( state ) =>
+		isFetchingCancellationOffers( state, purchaseId )
+	);
+
+	useEffect( () => {
+		if ( siteId && fetchingCancellationOffers === null ) {
+			dispatch( fetchCancellationOffers( siteId, purchaseId ) );
+		}
+	}, [ dispatch, fetchingCancellationOffers, siteId ] );
```

Comment:
`purchaseId` is not included in dependencies of this hook even though it is used inside